push-notifications/ios/Plugin/PushNotificationsDelegate.swift

```diff
@@ -0,0 +1,91 @@
+import Capacitor
+import UserNotifications
+
+public class PushNotificationsDelegate: NSObject, NotificationHandlerProtocol {
```

Comment:
I think this class name should be revisited. The term `delegate` has a lot of connotations which don't necessarily apply and can cause confusion. The protocol it adopts is a `handler` protocol not a `delegate` protocol, after all. `PushNotificationHandler` perhaps?